test/validation/api/buffer/buffer.c

```diff
@@ -370,6 +376,56 @@ static void buffer_test_pool_max_pools(void)
 		CU_ASSERT(odp_pool_destroy(pool[i]) == 0);
 }
 
+static void buffer_test_pool_alloc_free(void)
+{
+	odp_pool_param_t param;
+
+	memcpy(&param, &default_param, sizeof(odp_pool_param_t));
+	test_pool_alloc_free(&param);
+	set_cache_size(&param);
+	test_pool_alloc_free(&param);
+}
+
+static void buffer_test_pool_alloc_free_multi(void)
+{
+	odp_pool_param_t param;
+
+	memcpy(&param, &default_param, sizeof(odp_pool_param_t));
+	test_pool_alloc_free_multi(&param);
+	set_cache_size(&param);
+	test_pool_alloc_free_multi(&param);
```

Comment:
Running the same function multiple times within a test case can make debugging failures a bit difficult as there may be no way of knowing which parameters were used based on the test output alone. So perhaps it would make sense to separate these into multiple tests.
E.g.
```
buffer_test_pool_alloc_free_multi
buffer_test_pool_alloc_free_multi_min_cache
buffer_test_pool_alloc_free_multi_max_cache
```
Same goes for all the test cases.